buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java

```diff
@@ -152,6 +178,13 @@ public void setExcludes(String... classes) {
 
     @TaskAction
     public void runThirdPartyAudit() throws IOException {
+        if (isAllowAllMissingClasses() && (false == missingClassExcludes.isEmpty())) {
+            getLogger().error("There should be no explicit missing classes listed " +
+                "when the task is configured to allow all, yet these are present:\n {}", formatClassList(missingClassExcludes));
```

Comment:
With what I suggested above, I believe this check can be done at configuration time instead of runtime

Reply:
with the new DSL it didn't seem right to have this limitation any more, so one can go from broad to specific now.

Reply:
You mean that one could have:

```
thirdPartyAudit {
  ignoreMissingClasses '*'
  ignoreMissingClasses 'foo.bar.Baz'
}
```

That seems odd. Why do we need that leniency?

Reply:
I'm not arguing that we need it, I don't even like having i, but it's something that Gradle generally offers. So from a Gradle perspective, having a property that when set to some specific value can no longer be changed is odd. I think that changing the way Gradle works too much makes the build harder to understand and work with because it increases the amount of specific knowledge required to work with or change it.